pennylane/hf/dipole.py

```diff
@@ -0,0 +1,308 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for computing the dipole moment.
+"""
+import autograd.numpy as anp
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.hf.basis_data import atomic_numbers
+from pennylane.hf.hamiltonian import _generate_qubit_operator, _return_pauli, simplify
+from pennylane.hf.matrices import moment_matrix
+
+
+def dipole_integrals(mol, core=None, active=None):
+    r"""Return a function that computes the dipole moment integrals in the molecular orbital basis.
```

Comment:
We are actually using a matrix for the dipole integrals over atomic orbitals for consistency with other matrices we need for SCF calculations. It might be confusing to use matrix elements here.